src/atoms/cc-badge.js

```diff
@@ -0,0 +1,150 @@
+import { css, html, LitElement } from 'lit-element';
+import { defaultThemeStyles } from '../styles/default-theme.js';
+
+/**
+ * @typedef {import('./types.js').Intent} Intent
+ * @typedef {import('./types.js').Weight} Weight
+ */
+
+/**
+ * A component to highlight a small chunk of text.
+ *
+ * @cssdisplay inline-flex
+ */
+export class CcBadge extends LitElement {
+  static get properties () {
+    return {
+      circle: { type: Boolean, reflect: true },
+      iconAlt: { type: String },
+      iconSrc: { type: String },
+      intent: { type: String, reflect: true },
+      text: { type: String },
+      weight: { type: String, reflect: true },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {Boolean} Sets the badge to a bubble style. Should only be used to display 1 or 2 digits figures. */
+    this.circle = false;
+
+    /** @type {String|null} Sets the `alt` attribute value on the `<img>` tag. Only use if the image conveys additional info compared to surrounding text. */
+    this.iconAlt = null;
+
+    /** @type {String|null} Sets the icon displayed on the left of the text inside the badge. */
+    this.iconSrc = null;
+
+    /** @type {Intent} Sets the accent color used for the badge. */
+    this.intent = 'neutral';
+
+    /** @type {String} Sets the text content of the badge. */
+    this.text = null;
+
+    /** @type {Weight} Sets the style of the badge depending on how much one wants it to stand out. */
+    this.weight = 'dimmed';
+  }
+
+  render () {
+    return html`
+      ${this.iconSrc != null ? html`
+        <img src=${this.iconSrc} alt=${this.iconAlt ?? ''}>
+      ` : ''}
+      <span>${this.text}</span>
+    `;
+  }
+
+  static get styles () {
+    return [
+      defaultThemeStyles,
+      // language=CSS
+      css`
+        :host {
+          align-items: center;
+          /* only used as fallback in case theme is not set */
+          background-color: #777;
+          border-radius: 1em;
+          box-sizing: border-box;
+          color: var(--color-text-default);
+          display: inline-flex;
+          font-size: 0.8em;
+          gap: 0.3em;
+          min-height: 1.7em;
```

Comment:
This is something I really don't like to do but I wanted to make sure badges have the same size when outlined (border) and not outlined (no border).

I am open to suggestions if you find a better solution.